packages/common/src/state/dotlottie-state-machine.ts

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright 2023 Design Barn Inc.
+ */
+
+import type { AnimationEventName } from 'lottie-web';
+import { createMachine, interpret } from 'xstate';
+
+import type { DotLottieElement, DotLottiePlayer } from '../dotlottie-player';
+import { createError, getKeyByValue } from '../utils';
+
+import type {
+  DotLottieState,
+  EventMap,
+  StateAnimationSettings,
+  StateSettings,
+  StateTransitionEvents,
+  Transitionable,
+  XState,
+  XStateMachine,
+  XStateTargetEvent,
+} from './dotlottie-state';
+import { XStateEvents, EVENT_MAP } from './dotlottie-state';
+
+export class DotLottieStateMachine {
+  protected activeMachineId: string = '';
+
+  protected _service: any;
+
+  protected _domListeners = new Map<string, () => void>();
+
+  protected _domElement: DotLottieElement | undefined;
+
+  protected _playerListers = new Map<AnimationEventName, () => void>();
+
+  protected _player: DotLottiePlayer;
+
+  protected _machineSchemas = new Map<string, XStateMachine>();
+
+  public constructor(schemas: DotLottieState[], player: DotLottiePlayer) {
+    this._player = player;
+    this._machineSchemas = this._transfromToXStateSchema(schemas);
```

Comment:
typo in method name: this._transfromToXStateSchema(schemas);

should be:

this._transformToXStateSchema